src/stories/2-frontend-engine/validation/validation.stories.tsx

```diff
@@ -1,123 +1,626 @@
+				"Indicates that a field is mandatory, submitting with empty values trigger a validation failure.",
+		},
+	},
+};
+
+export const SoftValidation: Story = {
+	args: {
+		info: {
+			rule: { required: true, errorMessage: "Field is required.", soft: true },
+			ruleName: "soft",
+			ruleDescription:
+				"Shows a warning alert instead of an error alert when validation fails. The form can still be submitted.",
+		},
+	},
+};
+
+export const Uinfin: Story = {
+	args: {
+		info: {
+			rule: { uinfin: true, errorMessage: "Invalid NRIC/FIN" },
+			ruleName: "uinfin",
+			ruleDescription: "Indicates that the value must be a valid NRIC/FIN.",
+		},
+		fields: ["maskedField", "textarea", "textField"],
+	},
+};
+
+export const Url: Story = {
+	args: {
+		info: {
+			rule: { url: true, errorMessage: "Invalid url" },
+			ruleName: "url",
+			ruleDescription: "Indicates that the value must be a valid url.",
+		},
+		fields: ["maskedField", "textarea", "textField"],
+	},
+};
+
+export const Uuid: Story = {
+	args: {
+		info: {
+			rule: { uuid: true, errorMessage: "Invalid uuid" },
+			ruleName: "uuid",
+			ruleDescription: "Indicates that the value must be a valid uuid.",
+		},
+		fields: ["maskedField", "textarea", "textField"],
+	},
+};
+
+export const WhenExact: Story = {
+	name: "Conditional (Exact value)",
+	args: {
+		info: {
+			rule: {
+				when: {
+					field2: {
+						is: "something",
+						then: [{ required: true, errorMessage: "Required when field 2 value is `something`." }],
+						otherwise: [
+							{ empty: true, errorMessage: "Must not be filled when field 2 value is not `something`." },
+						],
 					},
-					sex: {
-						uiType: "select",
-						label: "Sex",
-						options: [
-							{ label: "Male", value: "male" },
-							{ label: "Female", value: "female" },
+				},
+			},
+			ruleName: "when",
+			ruleDescription:
+				"Applies validation to a field when another field is equal/not equal to a specific value according to the <code>is</code> value.",
+		},
+		excludeFields: OBJECT_BASED_VALIDATION_DEMO_FIELD_IDS,
+		overrides: [
+			{
+				for: ALL_VALIDATION_DEMO_FIELD_IDS,
+				schema: { field2: Object.assign({ ...VALIDATION_DEMO_CONFIGS.textField }, { label: "Field 2" }) },
+			},
+		],
+	},
+};
+
+export const WhenSchema: Story = {
+	name: "Conditional (Schema-based)",
+	args: {
+		info: {
+			rule: {
+				when: {
+					field2: {
+						is: [{ filled: true }, { min: 3 }],
+						then: [{ required: true, errorMessage: "Required when field 2 value is `something`." }],
+						otherwise: [
+							{ empty: true, errorMessage: "Must not be filled when field 2 value is not `something`." },
 						],
-						validation: [{ required: true, soft: true, errorMessage: "This field is optional" }],
 					},
-					...SUBMIT_BUTTON_SCHEMA,
 				},
 			},
+			ruleName: "when",
+			ruleDescription:
+				"Applies validation to a field when another field fulfils / fails the validation schema according to the <code>is</code> value.",
```

Comment:
small suggestion: would be helpful to mention the `is` rule is based on the conditional rendering rules, as it's not obvious what the keys are unless checking the typings

Reply:
just to be sure, you are referring to conditional validation rules and not conditional rendering rules right?